src/shared/user-record.c

```diff
@@ -1646,6 +1653,9 @@ int user_record_load(UserRecord *h, sd_json_variant *v, UserRecordLoadFlags load
                 { "pkcs11TokenUri",             SD_JSON_VARIANT_ARRAY,         dispatch_pkcs11_uri_array,            offsetof(UserRecord, pkcs11_token_uri),              0              },
                 { "fido2HmacCredential",        SD_JSON_VARIANT_ARRAY,         dispatch_fido2_hmac_credential_array, 0,                                                   0              },
                 { "recoveryKeyType",            SD_JSON_VARIANT_ARRAY,         sd_json_dispatch_strv,                offsetof(UserRecord, recovery_key_type),             0              },
+                { "selfModifiableFields",       JSON_VARIANT_ARRAY,            json_dispatch_strv,                   offsetof(UserRecord, self_modifiable_fields),        JSON_SAFE      },
+                { "selfModifiableBlobs",        JSON_VARIANT_ARRAY,            json_dispatch_strv,                   offsetof(UserRecord, self_modifiable_blobs),         JSON_SAFE      },
+                { "selfModifiablePrivileged",   JSON_VARIANT_ARRAY,            json_dispatch_strv,                   offsetof(UserRecord, self_modifiable_privileged),    JSON_SAFE      },
```

Comment:
this should be SD_JSON_VARIANT_ARRAY, not JSON_VARIANT_ARRAY. i.e. the correction was squashed into the 2nd commit, but should be here in the first